test/clojure/nrepl/bencode_test.clj

```diff
@@ -206,3 +205,31 @@
       (is (thrown? IllegalArgumentException
                    (write-bencode out {"obj" (Object.)})))
       (is (= "d3:obj" (String. (.toByteArray out)))))))
+
+(def valid-bencode-input-generator
+  (gen/recursive-gen #(gen/one-of [%
+                                   (gen/vector % 0 20)
+                                   (gen/map gen/string % {:min-elements 0
+                                                          :max-elements 20})])
+                     (gen/one-of [gen/string gen/large-integer])))
+
+(defspec generative-roundtrip-test {:num-tests 200}
+  (tc.prop/for-all
+   [value valid-bencode-input-generator]
+   (= value (>input (>output value :writer write-bencode) :reader read-bencode))))
+
+(comment
```

Comment:
Same for the performance tests. (e.g. something `### Performance tests`) and couple of sentences explaining them.

Reply:
Done